Replace FSLeyes for ITK-SNAP for manual segmentations (#34)
sct-pipeline/ukbiobank-spinalcord-csa: README.md

OPTIONAL:

* [ITK-SNAP](http://www.itksnap.org/pmwiki/pmwiki.php?n=Downloads.SNAP3) for correcting cord segmentations

    **NOTE:** 

>     Make sure to add SNAP to the system path:

Reviewer: ```suggestion
    Make sure to add ITK-SNAP to the system path:
```

    - For Windows, select the option during installation.

>     - For macOS, after installation, got to **Help->Install Command-Line Tools**.

Reviewer: ```suggestion
    - For macOS, after installation, go to **Help->Install Command-Line Tools**.
```

#### 1. Assess quality of segmentation and vertebral labeling
If segmentation or labeling issues are noticed while checking the quality report, proceed to manual segmentation correction or manual labeling of C2-C3 intervertebral disc at the posterior tip of the disc using the procedure below:

1. Create two .yml files that list the data to correct, one for segmentation and the other for vertebral labeling.
2. In QC report, search for "deepseg" to only display results of spinal cord segmentation, search for "vertebrae" to only display vertebral labeling.
3. Review segmentation and spinal cord labeling, note that the segmentation et vertebral labeling need to be accurate only between C2-C3, for cord CSA. 
4. If *major* issues are detected for C2-C3 segmentation and vertebral labeling, add the image's name into the corresponding .yml file as in the example below:

*.yml list for correcting cord segmentation:*
~~~
FILES_SEG:
- sub-1000032_T1w.nii.gz
- sub-1000083_T2w.nii.gz
~~~

> *.yml list for correcting vertebtal labeling:*

Reviewer: ```suggestion
*.yml list for correcting vertebral labeling:*
```

~~~
FILES_LABEL:
- sub-1000032_T1w.nii.gz
- sub-1000710_T1w.nii.gz
~~~

* `FILES_SEG`: Images associated with spinal cord segmentation
* `FILES_LABEL` Images associated with vertebral labeling (T1w images only)

> For the next steps, the script `uk_manual_correction` loops through all the files listed in .yml file and opens an interactive window to either correct manually segmentation or vertebral labeling. Each manually corrected label is saved under `derivative/labels/`folder at the root of `PATH_DATA` according to the BIDS convention. Each manually-corrected file has the suffix `-manual`. The procedure is described bellow for cord segmentation and for vertebral labeling.

Reviewer: ```suggestion
For the next steps, the script `uk_manual_correction` loops through all the files listed in .yml file and opens an interactive window to either manually correct segmentation or vertebral labeling. Each manually-corrected label is saved under `derivatives/labels/` folder at the root of `PATH_DATA` according to the BIDS convention. Each manually-corrected file has the suffix `-manual`. The procedure is described bellow for cord segmentation and for vertebral labeling.
```

#### 2. Correct segmentations
For manual segmentation, you will need ITK-SNAP and this repository only. See **[installation](#installation)** instructions and **[dependencies](#dependencies)**.

**TODO: put video here**

Run the following line and specify the .yml list for cord segmentation with the flag `-config`:
~~~
uk_manual_correction -config <.yml file> -path-in ~/ukbiobank_results/data_processed -path-out <PATH_DATA>
~~~

> After all correction are done, you can generate a QC report by adding the flag `-qc-only-` to the command above. Note that SCT is required for generating QC report.

Reviewer: ```suggestion
After all corrections are done, you can generate a QC report by adding the flag `-qc-only-` to the command above. Note that SCT is required for generating QC report.
```

#### 3. Vertebral labeling
Note that manual labeling uses SCT and the QC report is generated automatically.

**TODO: put video here**

Run the following line and specify the .yml list for vertebral labeling with the flag `-config`:
~~~
uk_manual_correction -config <.yml file> -path-in ~/ukbiobank_results/data_processed -path-out <PATH_DATA>
~~~

C2-C3 disc label will be located at the posterior tip of the disc as shown in the following image. 

![alt text](https://user-images.githubusercontent.com/2482071/100895704-dabf4a00-348b-11eb-8b1c-67d5024bfeda.png)

#### Upload the manually-corrected files

A QC report of the manually corrected files is created in a zip file. To update the dataset, add all manually-corrected files `derivatives/labels/`,  and include the qc zip file in the body of the PR. See our [internal procedure](https://github.com/neuropoly/data-management/blob/master/internal-server.md#upload) for more details.
**TODO see if it is possible to inlcude the qc zipe file in PR** 

#### Re-run the analysis

> After all the necessary segmentation and labels are corrected, re-run the analysis (`sct_run_batch`command in [Processing](###processing)). If manually-corrected files exists, they will be used intead of proceeding to automatic segmentation and labeling. Make sure to put the output results in another folder (flag `-path-output`) if you don't want the previous relsults to be overwritten. 

Reviewer: ```suggestion
After all the necessary segmentation and labels are corrected, re-run the analysis (`sct_run_batch` command in [Processing](###processing)). If manually-corrected files exist, they will be used instead of proceeding to automatic segmentation and labeling. Make sure to put the output results in another folder (flag `-path-output`) if you don't want the previous results to be overwritten. 
```
